### PR TITLE
Feature/adding tables to db

### PR DESCRIPTION
**Pull request by MartinMarkov05**

### CODE REVIEW

File: build/resources/main/hibernate.cfg.xml [hunk 1]
```diff
@@ -31,7 +31,10 @@
         </property>
         <!-- List of XML mapping files -->
         <mapping class= "com.vso.model.entity.User"/>
-
+        <mapping class= "com.vso.model.entity.Friendships"/>
+        <mapping class= "com.vso.model.entity.Photos"/>
+        <mapping class= "com.vso.model.entity.Post"/>
+        <mapping class= "com.vso.model.entity.likes"/>
```
Comment (ndim97) [inline build/resources/main/hibernate.cfg.xml:37]: Delete this row. You do not need class Likes

File: src/main/java/com/vso/model/entity/Friendships.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package com.vso.model.entity;
+
+import jakarta.persistence.*;
+import org.hibernate.annotations.Cascade;
+
+import java.lang.annotation.Target;
+
+@Entity
+@Table(name = "friendships")
+public class Friendships {
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/Friendships.java:10]: Rename this class to be Friendship, without s

File: src/main/java/com/vso/model/entity/Friendships.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    @ManyToOne(targetEntity = User.class)
+    @Cascade(org.hibernate.annotations.CascadeType.SAVE_UPDATE)
+    @JoinColumn(name = "user_id",insertable = false,updatable = false)
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/Friendships.java:14]: Here name = user_id_requester. Why you added these insertable = false,updatable = false ?

File: src/main/java/com/vso/model/entity/Friendships.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    private User requester;
+
+    @ManyToOne(targetEntity = User.class)
+    @Cascade(org.hibernate.annotations.CascadeType.SAVE_UPDATE)
+    @JoinColumn(name = "user_id",insertable = false,updatable = false)
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/Friendships.java:19]: The same question and change this: name = user_id_requested

File: src/main/java/com/vso/model/entity/Friendships.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    private User requested;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column
+    private long id;
+    @Column
+    private long user_id_requester;
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/Friendships.java:27]: You can delete this field.

File: src/main/java/com/vso/model/entity/Friendships.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    @Column
+    private long user_id_requested;
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/Friendships.java:29]: You can delete this field too.

File: src/main/java/com/vso/model/entity/Friendships.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    @Column
+    private String friendship_status;
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/Friendships.java:31]: This friendship_status is better to be an Enum class. Discuss this with your team.

File: src/main/java/com/vso/model/entity/Photos.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
+package com.vso.model.entity;
+
+import jakarta.persistence.*;
+import org.hibernate.annotations.Cascade;
+
+@Entity
+@Table(name = "photos")
+public class Photos {
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/Photos.java:8]: Rename this class to be Photo, without s

File: build/resources/main/hibernate.cfg.xml [hunk 9]
```diff
@@ -31,7 +31,10 @@
         </property>
         <!-- List of XML mapping files -->
         <mapping class= "com.vso.model.entity.User"/>
-
+        <mapping class= "com.vso.model.entity.Friendships"/>
+        <mapping class= "com.vso.model.entity.Photos"/>
```
Comment (ndim97) [inline build/resources/main/hibernate.cfg.xml:35]: After renaming classes change their names here too.

File: src/main/java/com/vso/model/entity/Post.java [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+@Table(name = "posts")
+public class Post {
+
+    @ManyToOne(targetEntity = User.class)
+    @Cascade(org.hibernate.annotations.CascadeType.SAVE_UPDATE)
+    @JoinColumn(name = "user_id")
+    private User user;
+
+    @OneToMany(targetEntity = likes.class,mappedBy = "post",cascade=CascadeType.ALL)
+    private List<likes> likesList;
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/Post.java:19]: These rows should be under privacy_status column, above the constructor

File: src/main/java/com/vso/model/entity/Post.java [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column
+    private long id;
+    @Column(insertable = false,updatable = false)
+    private int user_id;
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/Post.java:27]: You can delete these column. Hibernate will automatically created it for you.

File: src/main/java/com/vso/model/entity/Post.java [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+    @Column
+    private Date date;
+    @Column
+    private String text;
+   // @OneToMany(targetEntity = Post.class,mappedBy = "id")
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/Post.java:32]: Delete this comment. //...

File: src/main/java/com/vso/model/entity/Post.java [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+    @Column
+    private int parent_post_id;
+    @Column
+    private Date reported;
+    @Column
+    private Date deleted;
+    @Column
+    private String privacy_status;
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/Post.java:40]: This privacy_status is better to be an Enum class. Discuss this with your team.

File: src/main/java/com/vso/model/entity/Post.java [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+    private long id;
+    @Column(insertable = false,updatable = false)
+    private int user_id;
+    @Column
+    private Date date;
+    @Column
+    private String text;
+   // @OneToMany(targetEntity = Post.class,mappedBy = "id")
+    @Column
+    private int parent_post_id;
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/Post.java:34]: Delete this field too.

File: src/main/java/com/vso/model/entity/Post.java [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+    @Column
+    private Date reported;
+    @Column
+    private Date deleted;
+    @Column
+    private String privacy_status;
+
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/Post.java:41]: Add this code here (above the constructor):

@ManyToOne
    @JoinColumn(name = "parent_post_id")
    private Post parentPost;

    @OneToMany(mappedBy = "parentPost", fetch = FetchType.EAGER)
    private List<Post> comments;

    @ManyToMany(targetEntity = User.class, fetch = FetchType.LAZY)
    @JoinTable(name = "likes",
    joinColumns = {@JoinColumn(name = "post_id")},
    inverseJoinColumns = {@JoinColumn(name = "user_id")})
    private List<User> likes;

Comment (ndim97) [reply src/main/java/com/vso/model/entity/Post.java:41]: This code will add automaticaly new table 'likes', so you will not need this class Likes anymore.

File: src/main/java/com/vso/model/entity/User.java [hunk 16]
```diff
@@ -1,13 +1,50 @@
 package com.vso.model.entity;
 
 import jakarta.persistence.*;
+<<<<<<< HEAD
+
+import java.util.List;
+=======
+>>>>>>> 656a14378d066b2a7af27ecaa4daf1228bdaf9ae
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/User.java:8]: Remove all these git stuff

File: src/main/java/com/vso/model/entity/User.java [hunk 17]
```diff
@@ -1,13 +1,50 @@
...
+    private List<Post> postList;
+
+    @OneToMany(targetEntity = Photos.class,mappedBy = "user",cascade=CascadeType.ALL)
+    private List<Photos> photosList;
+
+    @OneToMany(targetEntity = Friendships.class,mappedBy = "requester",cascade=CascadeType.ALL )
+    private List<User> requestersList;
+
+    @OneToMany(targetEntity = Friendships.class,mappedBy = "requested",cascade=CascadeType.ALL)
+    private List<User> requestedList;
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/User.java:24]: These rows should be under the last column field and above the constructor.

File: src/main/java/com/vso/model/entity/User.java [hunk 18]
```diff
@@ -1,13 +1,50 @@
...
+
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     @Column
+<<<<<<< HEAD
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/User.java:29]: Remove this too.

File: src/main/java/com/vso/model/entity/User.java [hunk 19]
```diff
@@ -1,13 +1,50 @@
...
+    private long id;
+    @Column
+    private String email;
+    @Column
+    private int pass;
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/User.java:34]: I prefer here to be password, not pass. Discuss this with your team.

File: src/main/java/com/vso/model/entity/User.java [hunk 20]
```diff
@@ -1,13 +1,50 @@
...
+    @Column
+    private String name;
+    @Column
+    private int age;
+    @Column
+    private String avatar_url;
+    @Column
+    private String cover_url;
+    @Column
+    private String profile_status;
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/User.java:44]: This can be enum too.

File: src/main/java/com/vso/model/entity/User.java [hunk 21]
```diff
@@ -1,13 +1,50 @@
...
+    @Column
+    private String profile_role;
+=======
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/User.java:47]: Delete =====

File: src/main/java/com/vso/model/entity/User.java [hunk 22]
```diff
@@ -32,6 +69,7 @@ public User(String email,String password,String name, int age) {
         this.name = name;
         this.age = age;
     }
+>>>>>>> 656a14378d066b2a7af27ecaa4daf1228bdaf9ae
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/User.java:72]: Delete >>>>>

File: src/main/java/com/vso/model/entity/likes.java [hunk 23]
```diff
@@ -0,0 +1,31 @@
+package com.vso.model.entity;
+
+import jakarta.persistence.*;
+import org.hibernate.annotations.Cascade;
+
+@Entity
+@Table(name = "likes")
+public class likes {
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/likes.java:8]: Remove all class

File: src/main/java/com/vso/model/entity/User.java [hunk 24]
```diff
@@ -1,13 +1,50 @@
...
+import java.util.List;
+=======
+>>>>>>> 656a14378d066b2a7af27ecaa4daf1228bdaf9ae
 
 @Entity
 @Table(name = "users")
 public class User {
+
+    @OneToMany(targetEntity = Post.class,mappedBy = "user",cascade=CascadeType.ALL)
+    private List<Post> postList;
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/User.java:15]: This field is better to be only posts, without List in name. The type is enough to be List.
private List<Post> posts;

File: src/main/java/com/vso/model/entity/User.java [hunk 25]
```diff
@@ -1,13 +1,50 @@
...
+
+    @OneToMany(targetEntity = Photos.class,mappedBy = "user",cascade=CascadeType.ALL)
+    private List<Photos> photosList;
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/User.java:18]: Here only photots without -List at the end.

File: src/main/java/com/vso/model/entity/User.java [hunk 26]
```diff
@@ -1,13 +1,50 @@
...
+
+    @OneToMany(targetEntity = Friendships.class,mappedBy = "requester",cascade=CascadeType.ALL )
+    private List<User> requestersList;
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/User.java:21]: the same

File: src/main/java/com/vso/model/entity/User.java (hunk 17 above)
Comment (ndim97) [inline src/main/java/com/vso/model/entity/User.java:24]: the same

File: src/main/java/com/vso/model/entity/Photos.java [hunk 27]
```diff
@@ -0,0 +1,35 @@
...
+
+    @ManyToOne(targetEntity = User.class)
+    @Cascade(org.hibernate.annotations.CascadeType.SAVE_UPDATE)
+    @JoinColumn(name = "user_id")
+    private User user;
```
Comment (ndim97) [inline src/main/java/com/vso/model/entity/Photos.java:13]: Put this code above the constructorq under the columns

File: src/main/java/com/vso/model/entity/Post.java (hunk 10 above)
Comment (ndim97) [inline src/main/java/com/vso/model/entity/Post.java:19]: You can get the number of likes or who liked the post with a query to a database, you do not need this.

File: build/resources/main/hibernate.cfg.xml (hunk 1 above)
Comment (ndim97) [reply build/resources/main/hibernate.cfg.xml:37]: kfdgjfhnjgkuh

**Comment by IvanStoev1** [comment]
comment

**Pull request merged**